src/main/java/com/academia/andruhovich/library/service/impl/OrderServiceImpl.java

```diff
@@ -0,0 +1,209 @@
+
+        orderRepository.deleteById(id);
+    }
+
+
+    private Order isNeedRecalculate(Order order) {
+        if (DRAFT.equals(order.getStatus())) {
+            return recalculatePriceAndHistory(order);
+        }
+        return order;
+    }
+
+    private void mapping(OrderRequestDto orderRequestDto, Order order, Map<Long, Integer> orderContent, List<Book> books) {
+        order.setHistory(createHistoryBooks(orderContent, books));
+        order.setTotalPrice(calculateTotalPrice(orderContent, books));
+        order.setUpdatedAt(DateHelper.currentDate());
+        order.setStatus(orderRequestDto.getStatus());
+    }
+
+    protected String createHistoryBooks(Map<Long, Integer> orderContent, List<Book> books) {
+        Set<OrderContentWrapper> responseOrderContentSet = books.stream()
+                .map(book -> getOrderContentWrapper(orderContent, book))
+                .collect(Collectors.toSet());
+
+        return jsonConverter.collectionToJsonString(responseOrderContentSet);
+    }
+
+    private OrderContentWrapper getOrderContentWrapper(Map<Long, Integer> orderContent, Book book) {
+        BookDto bookDto = bookMapper.modelToDto(book);
+        Integer quantity = orderContent.get(book.getId());
+        return new OrderContentWrapper(bookDto, quantity);
+    }
+
+    protected BigDecimal calculateTotalPrice(Map<Long, Integer> orderContent, List<Book> books) {
+        return books.stream()
+                .map(book -> getBigDecimal(orderContent, book))
+                .reduce(BigDecimal.ZERO, BigDecimal::add);
+    }
+
+    private BigDecimal getBigDecimal(Map<Long, Integer> orderContent, Book book) {
+        BigDecimal quantity = BigDecimal.valueOf(orderContent.get(book.getId()));
+        return book.getPrice().multiply(quantity);
+    }
+
+    protected BigDecimal calculateTotalPrice(Set<OrderContentWrapper> orderContent, List<Book> books) {   //refactoring
+        return books.stream()
+                .map(book -> getBookSetBigDecimalPrice(orderContent, book))
+                .reduce(BigDecimal.ZERO, BigDecimal::add);
+    }
+
+    private BigDecimal getBookSetBigDecimalPrice(Set<OrderContentWrapper> orderContent, Book book) {
+        OrderContentWrapper currentBookAndQuantity = orderContent.stream()
+                .filter(element -> book.getId().equals(element.getBook().getId()))
+                .findFirst()
+                .orElseThrow(() ->
+                        new ResourceNotFoundException(String.format(BOOK_NOT_FOUND, book.getId())));
+        Integer quantity = currentBookAndQuantity.getQuantity();
+        BigDecimal quantityBigDecimal = BigDecimal.valueOf(quantity);
+        return book.getPrice().multiply(quantityBigDecimal);
+    }
+
+    protected Order recalculatePriceAndHistory(Order order) {
+        if (order.getHistory().startsWith("\"")) {
+            order.setHistory(order.getHistory().replace("\\", ""));
```

Comment:
move to sub method +add test
           if (order.getHistory().startsWith("\"")) {
            order.setHistory(order.getHistory().replace("\\", ""));
            order.setHistory(order.getHistory().substring(1, order.getHistory().length() - 1));
        }

Reply:
Refactored it and added test